app/web/components/LanguagePickerSelect.tsx

```diff
@@ -0,0 +1,195 @@
+import CheckIcon from "@mui/icons-material/Check";
+import ExpandMoreOutlinedIcon from "@mui/icons-material/ExpandMoreOutlined";
+import {
+  Box,
+  FormControl,
+  ListItemIcon,
+  ListItemText,
+  MenuItem,
+  Select as MuiSelect,
+  SelectChangeEvent,
+  Stack,
+  TextField,
+  useMediaQuery,
+  useTheme,
+} from "@mui/material";
+import { LANGUAGE_MAP } from "i18n/constants";
+import { useState } from "react";
+
+type LanguagePickerSelectProps = {
+  defaultValue?: string;
+  value?: string;
+  onSelect?: (value: string) => void;
+  displayMode?: "round" | "rect";
+};
+
+// note: manually add a non-secure cookie in development to test this functionality
+// TODO: this function lives in two places... where should it live?
+function getLangCookie() {
+  const name = "couchers-preferred-language=";
+  const decodedCookie = decodeURIComponent(document.cookie);
+  const ca = decodedCookie.split(";");
+  for (let i = 0; i < ca.length; i++) {
+    let c = ca[i];
+    while (c.charAt(0) == " ") {
+      c = c.substring(1);
+    }
+    if (c.indexOf(name) == 0) {
+      return c.substring(name.length, c.length);
+    }
+  }
+  return "";
+}
```

Comment:
Just curious a bit the logic here. Might be good in comments to give an example of the format of the cookie so it's a bit clearer.

Also would suggest rather than using `c`, `ca` etc for variable names, make them more descriptive so someone can look at it and know what it's doing.